src/platform/terminals/codeExecution/codeExecutionHelper.unit.test.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as assert from 'assert';
+import * as TypeMoq from 'typemoq';
+import { CodeExecutionHelperBase } from './codeExecutionHelper';
+import { MockEditor } from '../../../test/datascience/mockTextEditor';
+import { IServiceContainer } from '../../ioc/types';
+import { Uri, Selection } from 'vscode';
+import { MockDocumentManager } from '../../../test/datascience/mockDocumentManager';
+import { createMockedDocument } from '../../../test/datascience/editor-integration/helpers';
+
+function initializeMockTextEditor(inputText: string, selection: Selection): MockEditor {
+    const file = Uri.file('test.py');
+    const mockDocument = createMockedDocument(inputText, file, 1, true);
+    const mockTextEditor = new MockEditor(new MockDocumentManager(), mockDocument);
+    mockTextEditor.selection = selection;
+    return mockTextEditor;
+}
+
+const inputText = `print(1)
+
+if (true):
+    print(2)
+    print(3)
+`;
```

Comment:
We can have tests for multi-line string, and also mix use of spaces and tabs.

Reply:
this does not support mixed tabs and spaces, neither does the current implementation